internal/errors/errors_test.go

```diff
@@ -0,0 +1,1407 @@
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrTimeoutParseFailed(test.args.timeout)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrServerNotFound(t *testing.T) {
+	type args struct {
+		name string
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return an ErrServerNotFound error when the name is not empty.",
+			args: args{
+				name: "gateway.vald.svc.cluster.local",
+			},
+			want: want{
+				errors.New("server gateway.vald.svc.cluster.local not found"),
+			},
+		},
+		{
+			name: "return an ErrServerNotFound error when the name is empty.",
+			args: args{
+				name: "",
+			},
+			want: want{
+				errors.New("server  not found"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(tt *testing.T) {
+			if test.beforeFunc != nil {
+				test.beforeFunc(test.args)
+			}
+			if test.afterFunc != nil {
+				defer test.afterFunc(test.args)
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := ErrServerNotFound(test.args.name)
+			if err := test.checkFunc(test.want, got); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func TestErrOptionFailed(t *testing.T) {
+	type args struct {
+		err error
+		ref reflect.Value
+	}
+	type want struct {
+		want error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+	defaultCheckFunc := func(w want, got error) error {
+		if !Is(got, w.want) {
+			return fmt.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+		}
+		return nil
+	}
+	tests := []test{
+		{
+			name: "return an ErrOptionFailed error when err and ref are not empty.",
+			args: args{
+				err: errors.New("option failed error"),
+				ref: func() reflect.Value {
+					var i int
+					return reflect.ValueOf(&i)
+				}(),
+			},
+			want: want{
+				errors.New("failed to setup option :\t: option failed error"),
+			},
+		},
+		{
+			name: "return an ErrOptionFailed error when err is empty and ref is not empty.",
+			args: args{
+				ref: func() reflect.Value {
+					var i int
+					return reflect.ValueOf(&i)
+				}(),
+			},
+			want: want{
+				errors.New("failed to setup option :\t"),
+			},
+		},
+		{
+			name: "return an ErrOptionFailed error when err is not empty and ref is empty.",
+			args: args{
+				err: errors.New("option failed error"),
+			},
+			want: want{
+				errors.New("failed to setup option :\t: option failed error"),
+			},
+		},
```

Comment:
Should I add pattern that `runtime.FuncForPC(ref.Pointer()).Name())`  is not empty ?
if I made a mistake, I am sorry.

Reply:
well, it would be better.
how should I create it?